### PR TITLE
Polished uploud section

### PR DESCRIPTION
**Pull request by mt-empty**
## Proposed changes

* modified some css, added hover effect
* replaced cloud image with SVG
* added isDragActive effect for the cloud image, changes colour when dragging
* warning when the button is clicked but no file is uploaded

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)
- [ ] DevOps related

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [project guidelines](https://github.com/Visual-Testing-for-Android-Apps/Project-guidelines) doc
- [x] I locally linted my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

## Screenshots

Before uploading a file
image

While uploading a file
image

After uploading a file
image

Warning when the button is clicked but no file is uploaded
image

### CODE REVIEW

**Review by idomo1 — CHANGES_REQUESTED** [review]
👍 Nice work
One bug with the behaviour is that the button still works while it is in the disabled state

File: src/components/mainPage/UploadSection.jsx [hunk 1]
```diff
@@ -57,14 +58,22 @@ const UploadSection = () => {
         .removeEventListener("captchaEvent", captchaListener);
   }, []);
 
+  // changes the visibility of the button depending on the state of files
+  useEffect(() => {
+    setBtnOpacity(files.length != 0 ? 1 : 0.4);
+    console.log(`${files.length} ${btnOpacity}`);
```
[the remark this answers is not in the document]
Comment (mt-empty) [reply src/components/mainPage/UploadSection.jsx]: The opacity thing is what you sent me via slack

File: src/components/mainPage/UploadSection.jsx [hunk 2]
```diff
@@ -57,14 +58,22 @@ const UploadSection = () => {
         .removeEventListener("captchaEvent", captchaListener);
   }, []);
 
+  // changes the visibility of the button depending on the state of files
+  useEffect(() => {
+    setBtnOpacity(files.length != 0 ? 1 : 0.4);
```
[the remark this answers is not in the document]
Comment (mt-empty) [reply src/components/mainPage/UploadSection.jsx]: Done

File: src/components/mainPage/cloud.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+
+const Cloud = (props) => {
```
[the remark this answers is not in the document]
Comment (mt-empty) [reply src/components/mainPage/cloud.jsx]: Done

File: src/components/mainPage/UploadSection.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (idomo1) [reply src/components/mainPage/UploadSection.jsx]: The console log I mean

File: src/components/mainPage/UploadSection.jsx [hunk 4]
```diff
@@ -57,14 +60,40 @@ const UploadSection = () => {
...
+    } else {
+      setBtnOpacity(1);
+      setUploadAlertState("hidden");
+    }
+  }, [files.length]);
+
+  // prevent button from working if no files are uploaded
+  const handleOnClick = (event) => {
+    if (files.length == 0) {
+      event.preventDefault(event);
```
Comment (idomo1) [inline src/components/mainPage/UploadSection.jsx:77]: `preventDefault()` doesn't take any arguments afaik

File: src/components/mainPage/cloudIcon.jsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+
+const CloudIcon = (props) => {
+  console.log(props.fill);
```
Comment (idomo1) [inline src/components/mainPage/cloudIcon.jsx:4]: This log isn't useful

File: src/components/mainPage/upload.css [hunk 6]
```diff
@@ -13,19 +13,24 @@
...
   padding: 3rem;
 }
 
-.cloud-img {
-  padding-bottom: 0.8rem;
+.box:hover {
+  border: 0.2rem solid darkblue;
+}
+
+.UploadAlert {
```
Comment (idomo1) [inline src/components/mainPage/upload.css:34]: CSS classes should be hyphen-delineated, not camelCase

**Review by idomo1 — APPROVED** [review]
👍

File: src/components/mainPage/UploadSection.jsx (hunk 4 above)
Comment (mt-empty) [reply src/components/mainPage/UploadSection.jsx:77]: what about the other `preventDefault`

https://github.com/Visual-Testing-for-Android-Apps/web/pull/28/files#diff-d2d523e96242a52e5a296514284a8c406de8c9eab11dddcec7465f82e812765fL25

Comment (idomo1) [reply src/components/mainPage/UploadSection.jsx:77]: Oh yeah, that one shouldn't pass args either

File: src/components/mainPage/upload.css [hunk 7]
```diff
@@ -13,19 +13,24 @@
...
   padding: 3rem;
 }
 
-.cloud-img {
-  padding-bottom: 0.8rem;
+.box:hover {
+  border: 0.2rem solid darkblue;
+}
+
+.Upload-alert {
```
Comment (idomo1) [inline src/components/mainPage/upload.css:34]: Should be all lower case

**Review by akshithP — APPROVED** [review]
This looks great👍

**Review by akshithP — APPROVED** [review]

**Pull request merged**